packages/core/nuxt-theme-module/theme/components/AppHeader.vue

```diff
@@ -104,14 +159,66 @@ export default {
       await loadWishlist();
     });
 
+    const handleSearchInput = (input) => term.value = input;
+
+    const closeSearchAndRemoveTerm = () => {
+      term.value = '';
+      searchOpen.value = false;
+      changeSearchTerm(term.value);
+    };
+
+    const handleSearch = async (e) => {
```

Comment:
just have a single debounced function that handles both search AND passing term to the url